.gitpod.yml

```diff
@@ -0,0 +1,44 @@
+image:
+  file: docker/dev-gitpod.Dockerfile
+
+# --------------------------------------------------------
+# exposing ports for liveserve
+ports:
+  - port: 5500
+    onOpen: notify
+
+vscode:
+  extensions:
+    - ms-python.python
+    - ritwickdey.liveserver
+
+tasks:
+  - init: |
+      # this needs to be a proj var
+      export PRE_COMMIT_HOME=/workspace/.precommit
+      [ ! -d venv ] && python -m venv venv
+      source venv/bin/activate
```

Comment:
I found that a virtual environment is required, or at least greatly simplifies installation, due to problems with one of `enum-tools` dependencies.  See Currently use of `enum_tools`. See https://github.com/sphinx-toolbox/sphinx-jinja2-compat/issues/2.

Reply:
I'm wondering if we could/should abuse the fact that python executables in a venv know via `site` that they should be using the venv. Even if you don't activate the venv. Probably the same goes for scripts like `pytest`. We could hack something together with symlinks, or just plop `venv/bin` to the front of `$PATH`, and it would _probably_ "just work".

Hmm, or can we edit `/home/gitpod/.bashrc` to activate the venv on login?

Reply:
Just came across https://github.com/gitpod-io/gitpod/issues/1555#issuecomment-635911024 saying in 2020

> In theory, most `.bashrc` customizations should be dropped into `/home/gitpod/.bashrc.d/` as separate files.

Reply:
Good thought.  I will do some testing.  I like the .bashrc path forward the best, if it works.  It would be the most transparent to the user.

Reply:
It seems to me that the shell already has a pyenv environment activated, named `gitpod`. Probably has to do with the `60-python` script in `.bashrc.d/`. I'm not sure how we should interact with that.

Reply:
> It seems to me that the shell already has a pyenv environment activated, named gitpod.

I don't see any pyenv virtual environment named gitpod.  `pyenv virtualenvs` returns nothing.  Or is it something else?  One important thing in gitpod workspaces is that only files in /workspace persist.  Using a regular virtual environment works well for this.  I'm not too familiar with how to use pyenv for virtual environments.

Reply:
PS the word gitpod is given twice in the prompt.  One is for the username the other for the branch name on my fork.

Reply:
> PS the word gitpod is given twice in the prompt. One is for the username the other for the branch name on my fork.

Sorry, that's what confused me. I saw the second "gitpod" in the prompt, saw .pyenv in the default `which python` and I saw pyenv things in `.bashrc.d/60-python`. I've never used pyenv, just jumped to conclusions.

Reply:
This isn't trivial, gitpod has several stages and these things don't persist or source when I expect them too.  I suspect that the `enum-tools` dependency is creating a headache in combination with the pyenv model in gitpod too.  pyenv does things I don't understand.

Reply:
Using .bashrc works, thanks for the suggestion! The key was to do it in the docker container.  If done in a gitpod task, it doesn't persist.  If this gets more complex, we could consider having a file that is copied into the container rather than constructing it line by line.  For two lines and a short docker file, I think the current status is fine.